tests/test_mpi.py

```diff
@@ -50,6 +50,31 @@ def test_partitioning_fewer_dims(self):
         }
         assert f.shape == expected[distributor.nprocs][distributor.myrank]
 
+    @pytest.mark.parallel(mode=[2, 4])
+    def test_partitioning_fewer_dims_timefunc(self):
+        """Test domain decomposition for Functions defined over a strict subset
+        of grid-decomposed dimensions."""
+        size_x, size_y = 16, 16
+        grid = Grid(shape=(size_x, size_y))
+        x, y = grid.dimensions
+
+        # A function with fewer dimensions that in `grid`
+        f = TimeFunction(
+            name='f',
+            grid=grid,
+            dimensions=(grid.time_dim, x,),
+            shape=(10, size_x,),
+        )
+
+        distributor = grid.distributor
+        expected = {  # nprocs -> [(rank0 shape), (rank1 shape), ...]
+            2: [(8,), (8,)],
+            4: [(8,), (8,), (8,), (8,)]
```

Comment:
shouldn't this be 4?

Reply:
Decomposition for 4 ranks will do 2x2, so the x size doesn't change.

Reply:
Yes, stupid me.

Reply:
@rhodrin @FabioLuporini how is handled the duplication in that case?

Reply:
Yeah good question!